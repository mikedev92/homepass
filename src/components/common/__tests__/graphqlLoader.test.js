import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';

jest.dontMock('../graphqlLoader');

describe('graphqlLoader', () => {
  it('matches snapshot', () => {
    /* eslint-disable global-require */
    const graphqlLoader = require('../graphqlLoader').default;
    const Stub = graphqlLoader(() => <View />);
    const testProps = [
      { data: { loading: true } },
      { data: { loading: false } },
      { data: { error: { this: 'IS SOME CRAZY ERROR' } } },
    ];
    testProps.forEach((props) => {
      const wrapper = shallow(
        <Stub {...props} />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});

