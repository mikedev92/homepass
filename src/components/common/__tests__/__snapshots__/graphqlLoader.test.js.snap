// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphqlLoader matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`graphqlLoader matches snapshot 2`] = `
<Component
  data={
    Object {
      "loading": false,
    }
  }
/>
`;

exports[`graphqlLoader matches snapshot 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Styled(Text)>
    {
      "this": "IS SOME CRAZY ERROR"
    }
  </Styled(Text)>
  <Styled(Button)
    block={true}
    light={true}
    onPress={[Function]}
    outline={true}
  >
    <Styled(Text)>
      retry
    </Styled(Text)>
  </Styled(Button)>
</View>
`;
